src/main/scala/rise/core/types/Type.scala

```diff
@@ -3,8 +3,29 @@ package rise.core.types
 import arithexpr.arithmetic.RangeAdd
 import rise.core._
 
-sealed trait Type {
-  override def hashCode(): Int = this match {
+object alphaEquiv {
+  val equiv : Type => Type => Boolean = a => b => {
```

Comment:
@Bastacyclop  @fedepiz I'm trying to abstract away type equality with alpha equivalence here, while still keeping the old behavior for the time being by having `Type.equals` actually call this. I've done something similar with alpha equivalence for terms, and that works, but this one here doesn't, and I cannot pin point why. Am I missing anything obvious?

Reply:
Got it! It was a missing case for pair types. Thinking of going for a more defensive programming style where the compiler can help with exhaustiveness checking.